src/test/kotlin/uk/gov/dluhc/printapi/logging/CorrelationIdMdcIntegrationTest.kt

```diff
@@ -0,0 +1,302 @@
+            val expectedRequestId = certificate.printRequests[0].requestId!!
+
+            val statusUpdateFile = "status-20220928235441000.json"
+            writeContentToRemoteOutBoundDirectory(
+                statusUpdateFile,
+                objectMapper.writeValueAsString(
+                    buildPrintResponses(
+                        batchResponses = emptyList(),
+                        printResponses = listOf(
+                            buildPrintResponse(
+                                requestId = expectedRequestId
+                            )
+                        )
+                    )
+                )
+            )
+
+            // When
+            processPrintResponsesBatchJob.pollAndProcessPrintResponses()
+
+            // Then
+            await.atMost(30, TimeUnit.SECONDS).untilAsserted {
+                val logEvent = TestLogAppender.getLogEventMatchingRegex(
+                    "Finding matching print responses from directory: \\[EROP/Dev/OutBound\\]",
+                    Level.INFO
+                )
+                assertThat(logEvent).isNotNull
+                val expectedCorrelationId = logEvent!!.mdcPropertyMap[CORRELATION_ID]
+
+                assertThat(
+                    TestLogAppender.getLogEventMatchingRegex(
+                        "Renaming \\[$statusUpdateFile\\] to \\[$statusUpdateFile.processing\\] in directory:\\[EROP/Dev/OutBound\\]",
+                        Level.INFO
+                    )
+                ).hasCorrelationId(expectedCorrelationId)
+                assertThat(
+                    TestLogAppender.getLogEventMatchingRegex(
+                        "Submitting SQS message for file: \\[1 of 1\\] with payload: ProcessPrintResponseFileMessage\\(directory=EROP/Dev/OutBound, fileName=$statusUpdateFile.processing\\)",
+                        Level.INFO
+                    )
+                ).hasCorrelationId(expectedCorrelationId)
+                assertThat(
+                    TestLogAppender.getLogEventMatchingRegex(
+                        "Begin processing PrintResponse file \\[$statusUpdateFile.processing\\] from directory \\[EROP/Dev/OutBound\\]",
+                        Level.INFO
+                    )
+                ).hasCorrelationId(expectedCorrelationId)
+                assertThat(
+                    TestLogAppender.getLogEventMatchingRegex(
+                        "Removing processed file \\[$statusUpdateFile.processing\\] from directory \\[EROP/Dev/OutBound\\]",
+                        Level.INFO
+                    )
+                ).hasCorrelationId(expectedCorrelationId)
+                assertThat(
+                    TestLogAppender.getLogEventMatchingRegex(
+                        "Begin processing PrintResponse with requestId $expectedRequestId",
+                        Level.INFO
+                    )
+                ).hasCorrelationId(expectedCorrelationId)
+            }
+        }
+    }
```

Comment:
Looks good 👍 

Only thing I would question is whether we are testing different things in above two tests - are they both variations of running batch job that submits to SQS listener and then processes the SQS message.  Interesting that the second flow includes log about sending SQS message so is probably most useful.

Reply:
Yeah, they are kinda testing the same thing in that the correlation ID is being passed between processes.
I agree, I think the 2nd test (ProcessPrintResponsesBatch) is possibly more useful because its going through more async processes:
ProcessPrintResponsesBatch is a scheduled job which sends a message to a queue. That message is picked up by it's listener, which in turn will send another message to another queue. That 2nd message is picket up by its listener to complete the processing.
So in this case its nice to see/assert the correlation ID has gone through 3 separate processes 👍 

The other test for BatchPrintRequests is basically the same, but there's only 1 queue involved. So its:
BatchPrintRequests is a scheduled job which sends a message to a queue. That message is picked up by it's listener to complete the processing.

I quite like having them both, if nothing else via their comments and assertions, to demonstrate and document what the correlation ID is supposed to do and what value it will have for us when trawling logs